src/main.cc

```diff
@@ -239,7 +241,7 @@ static int daemon_main(LiteClient& client, const bpo::variables_map& variables_m
     LOG_INFO << "Active Target: " << current.filename() << ", sha256: " << current.sha256Hash();
     LOG_INFO << "Checking for a new Target...";
 
-    if (!client.checkForUpdates()) {
+    if (!client.checkForUpdatesBegin()) {
       LOG_WARNING << "Unable to update latest metadata, going to sleep for " << interval
                   << " seconds before starting a new update cycle";
       std::this_thread::sleep_for(std::chrono::seconds(interval));
```

Comment:
nope, if the `begin` is not successful then `end` is not needed, the post callback is sent here https://github.com/foundriesio/aktualizr-lite/pull/90/files#diff-27b40ef43c61478035883e6eee1dd2f6d13fdd1d20061e0dc1edcaf4c3d06dcdR177. But, I can change it if you think it will be better to have explicit `end` in the case of `begin` failure.

Reply:
@doanac if `begin` fails then `end` is not needed, otherwise the `begin` needs to return an error context which is useless in the positive case.